modules/configuring-quay-rosa-aws-sts.adoc

```diff
@@ -0,0 +1,140 @@
+----
++
+.Example output
++
+[source,terminal]
+----
+oidc.op1.openshiftapps.com/123456
+----
+
+. On the Identity and Access Management (IAM) console of the Amazon Web Services (AWS) console:
+
+.. Click *Roles*.
+
+.. Click *Custom Trust Policy*, which shows an editable JSON policy.
+
+.. Under the `Principal` parameter, add `Federated:<your_user_ARN>:<serviceAccountIssuer_domain_path>`. For example:
++
+[source,json]
+----
+        	"Principal": {
+            	"Federated": "arn:aws:iam::123456:oidc-provider/oidc.op1.openshiftapps.com/123456"
+        	},
+----
+
+.. Under the `Condition` parameter, add `StringEquals”: “<serviceAccountIssuer>:sub”: “system:serviceAccount:<quay_namespace>:<quay_registry_using_serviceAccount>`. For example:
++
+[source,json]
+----
+        	"Condition": {
+            	"StringEquals": {
+                	"oidc.op1.openshiftapps.com/123456:sub": "system:serviceaccount:<quay_enterprise>:<quay_registry_using_serviceAccount>"
+            	}
+        	}
+----
+
+.. Verify that your User ARN is configured correct, then click *Next*. You are prompted to select permissions to add to the role. 
+
+.. After you have selected the appropriate permissions, click *Next*. Then, provide your role a name, verify your configuration, and click *Create Role*.
+
+.. On the *Roles* page, click the new role and copy the `Role ARN` resource. For example:
++
+[source,text]
+----
+arn:aws:iam::123456:role/test_s3_access
+----
+
+. On the {productname} web console:
+
+.. Click *Operators* -> *Installed Operators*. 
+
+.. Click *Red Hat Quay*. 
+
+.. Click *Quay Registry* and then the name of your {productname} registry. 
+
+.. Under *Config Bundle Secret*, click the name of your registry configuration bundle, for example, *quay-registry-config-bundle-12345*. 
+
+.. On the configuration bundle page, click *Actions* to reveal a drop-down menu. Then click *Edit Secret*. 
+
+.. Update your the `DISTRIBUTED_STORAGE_CONFIG` fields of your `config.yaml` file with the following information:
++
+[source,yaml]
+----
+# ...
+DISTRIBUTED_STORAGE_CONFIG:
+   default:
+    - STSS3Storage
+    - sts_role_arn: <role_arn> <1>
+      s3_bucket: <s3_bucket_name> <2>
+      storage_path: <storage_path> <3> 
+      s3_region: <region> <4>
+# ...
+----
+<1> The unique Amazon Resource Name (ARN) required when configuring AWS STS
+<2> The name of your s3 bucket.
+<3> The storage path for data. Usually `/datastorage`.
+<4> Optional. The Amazon Web Services region. Defaults to `us-east-1`.
+
+. Click *Save*. Your `QuayRegistry` custom resource (CR) automatically restarts.
+
+. Annotate the Service Account (SA) that executes pods with the EKS configuration values. For example:
++
+[source,terminal]
+----
+$ 
+oc annotate sa registry-quay-app "eks.amazonaws.com/role-arn"="arn:aws:iam::814174734718:role/test_s3_access" "eks.amazonaws.com/audience"="sts.amazonaws.com" "eks.amazonaws.com/sts-regional-endpoints"="true"
```

Comment:
should align with above role arn value, please modify like:
`$ oc annotate sa registry-quay-app "eks.amazonaws.com/role-arn"="arn:aws:iam::123456:role/test_s3_access" "eks.amazonaws.com/audience"="sts.amazonaws.com" "eks.amazonaws.com/sts-regional-endpoints"="true"`